### PR TITLE
Optional component definition

### PR DESCRIPTION
**Pull request by benzwreck**
Makes `component` defined in yaml file optional. It is not always required, e.g. when only primitive values are used in `paths` definition.

### CODE REVIEW

**Comment by adamw** [comment]
There are conflicts to resolve

File: openapi-codegen/core/src/main/scala/sttp/tapir/codegen/openapi/models/OpenapiComponent.scala [hunk 1]
```diff
@@ -5,11 +5,14 @@ case class OpenapiComponent(schemas: Map[String, OpenapiSchemaType])
 object OpenapiComponent {
   import io.circe._
 
-  implicit val OpenapiComponentDecoder: Decoder[OpenapiComponent] = { (c: HCursor) =>
+  implicit val OpenapiComponentDecoder: Decoder[Option[OpenapiComponent]] = { (c: HCursor) =>
```
Comment (kciesielski) [inline openapi-codegen/core/src/main/scala/sttp/tapir/codegen/openapi/models/OpenapiComponent.scala:8]: **question:** Shouldn't it be enough to keep the `OpenapiComponent` decoder? The `Option[]` wrapper should be then derived by circe?

Comment (kciesielski) [reply openapi-codegen/core/src/main/scala/sttp/tapir/codegen/openapi/models/OpenapiComponent.scala:8]: I just tried it with this file unchanged, and tests pass :)

**Comment by adamw** [comment]
Thanks!

**Pull request merged**